spec/services/search/tag_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+
+RSpec.describe Search::Tag, type: :service, elasticsearch: true do
+  describe "::index" do
+    it "indexes a tag to elasticsearch" do
+      tag = FactoryBot.create(:tag)
+      expect { described_class.get(tag.id) }.to raise_error(Elasticsearch::Transport::Transport::Errors::NotFound)
+      described_class.index(tag.id, id: tag.id)
```

Comment:
not sure I understood this test, why pass the index twice?

Reply:
> why pass the index twice?

What do you mean by this? We check for the tag, confirm its not there in the index. Then we index it once, then we check for its presence again to confirm it is there.

Reply:
sorry, wrong question: "why pass the ID twice" :D

Reply:
Ah, great question and I thought this might get confusing but I left it bc this will change once I introduce serialization to the model in the next PR. Here is what is actually happening here
```ruby
serialized_es_data =  { id: tag.id }
described_class.index(tag.id, serialized_es_data)
```
Once we start serializing tags this [will change to this](https://github.com/thepracticaldev/dev.to/compare/mstruve/index-tags-to-es#diff-0049da0ae8599d807a6ddbe9700267ccR8)
```ruby
described_class.index(tag.id, tag.serialized_search_hash)
```